### PR TITLE
Backport 14682

### PR DESCRIPTION
**Pull request by nickvergessen**
Backport of #14682  and #14685 

@DeepDiver1975 @MorrisJobke @PVince81 

Backport approved: https://github.com/owncloud/core/pull/14682#issuecomment-77120663

### CODE REVIEW

**Comment by nickvergessen** [comment]
Remember stable8 is frozen

File: tests/lib/files/objectstore/noopscanner.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+		$this->assertEquals(
+			[],
+			$this->scanner->scan('', \OC\Files\Cache\Scanner::SCAN_SHALLOW),
+			'Asserting that no error occurred while scan(SCAN_SHALLOW)'
+		);
+
+		$this->scanner->backgroundScan();
+
+		$this->scanner->backgroundScan();
```
Comment (nickvergessen) [inline tests/lib/files/objectstore/noopscanner.php]: https://github.com/owncloud/core/pull/14685#discussion_r25765480

**Comment by nickvergessen** [comment]
@owncloud-bot retest this please

**Comment by DeepDiver1975** [comment]
@nickvergessen 

````
00:02:25.601 .............................................................  915 / 2927 ( 31%)
00:02:29.242 ..............PHP Fatal error:  Access level to OC\Files\ObjectStore\NoopScanner::scanChildren() must be public (as in class OC\Files\Cache\Scanner) in /var/jenkins/workspace/pull-request-analyser-ng-simple@3/label/SLAVE/lib/private/files/objectstore/noopscanner.php on line 75
00:02:29.629 PHP Stack trace:
00:02:29.629 PHP   1. {main}() /usr/local/bin/phpunit:0
00:02:29.629 PHP   2. PHPUnit_TextUI_Command::main() /usr/local/bin/phpunit:612
00:02:29.630 PHP   3. PHPUnit_TextUI_Command->run() phar:///usr/local/bin/phpunit/phpunit/TextUI/Command.php:138
00:02:29.630 PHP   4. PHPUnit_TextUI_TestRunner->doRun() phar:///usr/local/bin/phpunit/phpunit/TextUI/Command.php:186
00:02:29.630 PHP   5. PHPUnit_Framework_TestSuite->run() phar:///usr/local/bin/phpunit/phpunit/TextUI/TestRunner.php:423
00:02:29.630 PHP   6. PHPUnit_Framework_TestSuite->run() phar:///usr/local/bin/phpunit/phpunit/Framework/TestSuite.php:751
00:02:29.630 PHP   7. PHPUnit_Framework_TestCase->run() phar:///usr/local/bin/phpunit/phpunit/Framework/TestSuite.php:751
00:02:29.631 PHP   8. PHPUnit_Framework_TestResult->run() phar:///usr/local/bin/phpunit/phpunit/Framework/TestCase.php:722
00:02:29.631 PHP   9. PHP_Invoker->invoke() phar:///usr/local/bin/phpunit/phpunit/Framework/TestResult.php:641
00:02:29.632 PHP  10. call_user_func_array:{phar:///usr/local/bin/phpunit/php-invoker/Invoker.php:93}() phar:///usr/local/bin/phpunit/php-invoker/Invoker.php:93
00:02:29.632 PHP  11. PHPUnit_Framework_TestCase->runBare() phar:///usr/local/bin/phpunit/php-invoker/Invoker.php:93
00:02:29.632 PHP  12. Test\Files\Cache\NoopScanner->setUp() phar:///usr/local/bin/phpunit/phpunit/Framework/TestCase.php:762
00:02:29.632 PHP  13. spl_autoload_call() phar:///usr/local/bin/phpunit/phpunit/Framework/TestCase.php:25
00:02:29.633 PHP  14. OC\Autoloader->load() phar:///usr/local/bin/phpunit/phpunit/Framework/TestCase.php:0
00:02:29.663 PHP Warning:  Unknown: Cannot destroy the zip context in Unknown on line 0
00:02:29.663 PHP Stack trace:
00:02:29.667 PHP   1. {main}() /usr/local/bin/phpunit:0
00:02:29.667 PHP   2. PHPUnit_TextUI_Command::main() /usr/local/bin/phpunit:612
00:02:29.667 PHP   3. PHPUnit_TextUI_Command->run() phar:///usr/local/bin/phpunit/phpunit/TextUI/Command.php:138
00:02:29.668 PHP   4. PHPUnit_TextUI_TestRunner->doRun() phar:///usr/local/bin/phpunit/phpunit/TextUI/Command.php:186
00:02:29.668 PHP   5. PHPUnit_Framework_TestSuite->run() phar:///usr/local/bin/phpunit/phpunit/TextUI/TestRunner.php:423
00:02:29.668 PHP   6. PHPUnit_Framework_TestSuite->run() phar:///usr/local/bin/phpunit/phpunit/Framework/TestSuite.php:751
00:02:29.668 PHP   7. PHPUnit_Framework_TestCase->run() phar:///usr/local/bin/phpunit/phpunit/Framework/TestSuite.php:751
00:02:29.668 PHP   8. PHPUnit_Framework_TestResult->run() phar:///usr/local/bin/phpunit/phpunit/Framework/TestCase.php:722
00:02:29.668 PHP   9. PHP_Invoker->invoke() phar:///usr/local/bin/phpunit/phpunit/Framework/TestResult.php:641
00:02:29.668 PHP  10. call_user_func_array:{phar:///usr/local/bin/phpunit/php-invoker/Invoker.php:93}() phar:///usr/local/bin/phpunit/php-invoker/Invoker.php:93
00:02:29.695 PHP  11. PHPUnit_Framework_TestCase->runBare() phar:///usr/local/bin/phpunit/php-invoker/Invoker.php:93
00:02:29.695 PHP  12. Test\Files\Cache\NoopScanner->setUp() phar:///usr/local/bin/phpunit/phpunit/Framework/TestCase.php:762
00:02:29.695 PHP  13. spl_autoload_call() phar:///usr/local/bin/phpunit/phpunit/Framework/TestCase.php:25
00:02:29.696 PHP  14. OC\Autoloader->load() phar:///usr/local/bin/phpunit/phpunit/Framework/TestCase.php:0
00:02:29.696 PHP Warning:  Unknown: Cannot destroy the zip context in Unknown on line 0

**Comment by nickvergessen** [comment]
yeah figured that aswell on the second run, will fix after 8.0.2 is open to merge into

**Comment by nickvergessen** [comment]
@DeepDiver1975 @MorrisJobke review so it can be merged once 8.0.3 is out?

**Comment by MorrisJobke** [comment]
:+1:

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**